content/chainguard/chainguard-images/working-with-images/incert-custom-certs.md

```diff
@@ -0,0 +1,269 @@
+
+```sh
+docker ps
+```
+```
+CONTAINER ID   IMAGE                      COMMAND                  CREATED         STATUS         PORTS                                       NAMES
+9e211033635b   cgr.dev/chainguard/nginx   "/usr/sbin/nginx -c …"   2 minutes ago   Up 2 minutes   0.0.0.0:8443->8443/tcp, :::8443->8443/tcp   agitated_jones
+```
+
+As this output shows, the name of the Nginx container in this example is `agitated_jones`. Replace this with the name of your own container in the following command:
+
+```sh
+docker inspect --format '{{ .NetworkSettings.IPAddress }}' agitated_jones
+```
+
+This will return the container's IP address:
+
+```
+172.17.0.2
+```
+
+Next, use Chainguard's `curl` image to attempt to reach the container. Be sure to replace `172.17.0.2` with your Nginx container's actual IP address, if different. 
+
+```sh
+docker run -it --add-host example.com:172.17.0.2 cgr.dev/chainguard/curl:latest-dev https://example.com:8443
+```
+
+> **Note**: You might have noticed that [example.com](https://example.com) is a real website. Instead of using the `curl` container to reach the actual `example.com`, this command includes the `--add-host` option to map the hostname `example.com` to the local IP address currently being used by the Nginx container. 
+
+However, the public Chainguard `curl` image doesn't have the certificate inside it, so this command will fail. 
+
+```
+curl: (60) rustls_connection_process_new_packets: invalid peer certificate: UnknownIssuer
+More details here: https://curl.se/docs/sslcerts.html
+
+curl failed to verify the legitimacy of the server and therefore could not
+establish a secure connection to it. To learn more about this situation and
+how to fix it, please visit the web page mentioned above.
+```
+
+The next step is to create an image that has our self-signed certificate built into it. For that, we'll use `incert`.
+
+
+## Using `incert` to insert a custom certificate into an image
+
+`incert` is a Go program from Chainguard that appends CA certificates to Docker images and pushes the modified image to a specified registry. This tool is still in active development, so feedback is welcome.
+
+Run the following command to build a new image using Chainguard's `curl` image as its base and insert the `selfsigned.pem` certificate into it. 
+
+```sh
+incert -ca-certs-file selfsigned.pem -platform linux/arm64 -image-url cgr.dev/chainguard/curl:latest -dest-image-url ttl.sh/curl-with-cert:1h
+```
+
+This command uses the `-ca-certs-file` option to specify that `incert` should use the `selfisgned.pem` certificate file; the `-platform` option to specify that it wants to build an image for `linux/arm64`; the `-image-url` option to specify the image we want to build on as our base image (here we specify Chainguard's `curl` image); and the `-dest-image-url` to pass the registry where we want the resulting image to be uploaded to.
```

Comment:
I guess it's worth mentioning they should change the -platform argument to match the host platform.

Reply:
good call!